lib/hpke/include/hpke/certificate.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "bytes/bytes.h"
+#include <memory>
+#include <optional>
+
+using namespace bytes_ns;
+using namespace std;
+
+namespace hpke {
+
+// Wrapper around X509 OpenSsl API
+struct Certificate
+{
+
+  struct Signature
+  {
+    enum struct ID : uint16_t
+    {
+      unknown = 0x0000,
+      ED25519 = 0x0001,
+      ED448 = 0x0002,
+    };
+
+    struct PublicKey
+    {
+      bytes data;
+    };
+
+    PublicKey pkey;
+    ID algorithm;
```

Comment:
```suggestion
    const ID public_key_algorithm;
```

... to distinguish it from the algorithm used to sign the cert.